plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -1609,6 +1608,30 @@ public static function get_product_usage_notice_rules() {
 		return $data;
 	}
 
+	/**
+	 * Verify request hash created by WCCOM.
+	 *
+	 * @param string $request_hash
+	 * @return bool
+	 */
+	public static function verify_request_hash( string $request_hash ): bool {
+		$request = WC_Helper_API::get(
+			'verify-request-hash',
+			array(
+				'authenticated' => true,
+				'query_string'  => '?request_hash=' . $request_hash,
+			)
+		);
+
+		if ( wp_remote_retrieve_response_code( $request ) !== 200 ) {
+			return false;
+		}
+
+		$data = json_decode( wp_remote_retrieve_body( $request ), true );
+
+		return $data['success'] === true;
```

Comment:
```suggestion
		return isset( $data['success'] ) && $data['success'] === true;
```

just in case WCCOM have some weird error the stores do not suddenly have warnings

Reply:
Resolved